test/validation.test.js

```diff
@@ -46,4 +63,105 @@ describe('dataset validation in frontend', function () {
     expect(status.tables).to.equal('OK')
     expect(status.graphs).to.equal('OK')
   })
+  it(`small invalid dataset with duplicated row on ${SPECSTORE}`, async function () {
+    this.timeout(600000)
+    const urlLatest = `${SPECSTORE}/${OWNERID}/${datasetsToTest[1].name}/latest`
+    let body = await apiStatus(urlLatest)
+    const revisionId = body.id.split('/').pop()
+    const revisionUrl = `${SPECSTORE}/${OWNERID}/${datasetsToTest[1].name}/${revisionId}`
+    body = await apiStatus(revisionUrl)
+    expect(body.spec_contents.meta.dataset).to.equal(`${datasetsToTest[1].name}`)
+    expect(body.state).to.equal('FAILED')
+  })
+  it(`small invalid dataset with duplicated row on ${DOMAIN}`, async function () {
+    this.timeout(1800000)
```

Comment:
Waw half hour timeout?

Reply:
It is temporary, it will be removed when I refactor to `ava`.